src/function/FunctionLink.ts

```diff
@@ -298,11 +306,179 @@ export function addLinkTools(
         WorkspaceTerms[WorkspaceElements[WorkspaceLinks[id].source].iri]
           .inScheme
       )
-      ].readOnly;
+    ].readOnly;
   let tools = [verticesTool, segmentsTool];
   if (!readOnly) tools.push(removeButton);
   let toolsView = new joint.dia.ToolsView({
-    tools: tools
+    tools: tools,
   });
   linkView.addTools(toolsView);
 }
+
+export async function getCacheConnections(iri: string) {
+  const terms: { [key: string]: any } = {};
+  const restrictions: { [key: string]: Restriction[] } = {};
+  const subClass: string[] = await fetchSubClasses(
+    AppSettings.contextEndpoint,
+    iri
+  );
+  const subClasses = subClass
+    .filter((subClass) => !(subClass in WorkspaceTerms))
+    .concat(
+      WorkspaceTerms[iri].subClassOf.filter(
+        (subClass) => !(subClass in WorkspaceTerms)
+      )
+    );
+  const relationships: {
+    relation: string;
+    target: string;
+    labels: { [key: string]: string };
+  }[] = [];
+  const connections: {
```

Comment:
It would be definitely better to specify complex types like this one outside of a function so that the types may be reused... and also it makes the function quite hard to read with so much going on.